Source/Directory.h

```diff
@@ -53,94 +24,148 @@ class Directory
      * description - description
      * subpath - again source dependent but used for finding the module
      */
-
+	
     /**
      * Opens a directory with the given URL, downloading the latest version of
      * the contents. 
-     */
-    Directory (URL location)
-    {
-        DownloadCache cache;
-        String file = cache.downloadTextFile (location);
-        ScopedPointer<XmlElement> xml = XmlDocument (file).getDocumentElement();
+	*/
+    Directory(URL location)
+	{
+		DownloadCache cache;
+		String file = cache.downloadFromDatabase();
 
-        if (xml)
+		if (!file.isEmpty())
         {
-            directory = ValueTree::fromXml (*xml);
+			directory = Database::parseToArray (file);
         }
-        else
+		else
         {
             throw JpmFatalExcepton ("directory format error or network problem",
                                     "Check " 
                                     + cache.getCachedFileLocation (location).getFullPathName() 
                                     + " for debugging which should contain the contents of " 
                                     + location.toString (true));
-        }
+		}
+	}
     }
```

Comment:
And jpm list crashes here.  If it works for you let me know and I'll
investigate here .... it could be this new linux box! The input stream
is nullptr ... probably needs to test for that and an error message
either way.  I can pop that in.  But let me know that it's working at
your end...

Program received signal SIGSEGV, Segmentation fault. 0x000000000040a732
in Database::get (this=0x7fffffffe010, endpoint=...) at
../../Source/Database.h:277 277        queryResult = in-
>readEntireStreamAsString(); (gdb) print in

On Thu, Dec 10, 2015, at 08:27 PM, Adam Wilson wrote:
> In Source/Directory.h[1]:


> >          {
>              throw JpmFatalExcepton ("directory format error or
>              network problem",                         "Check "
>                                      + cache.getCachedFileLocation
>                                        (location).getFullPathName()
>                                      + " for debugging which should
>                                        contain the contents of "
>                                      + location.toString (true));
> -        }
> +             }
> +     } } Sorry, thought I’d committed those fixes, try again now. …
>       On 10 Dec 2015, at 20:07, Jim <notifications@github.com> wrote:

 In Source/Directory.h <#1 (comment)[2]>:

 >          {    throw JpmFatalExcepton ("directory format error or
 >          network problem",                            "Check "
 >                                      + cache.getCachedFileLocation
 >                                        (location).getFullPathName()
 >                                      + " for debugging which should
 >                                        contain the contents of "
 >                                      + location.toString (true));
 > -        }
 > +            }
 > +    } }
 This might be an error? I think this is the cause of one of the
 build errors?

 — Reply to this email directly or view it on GitHub
 <https://github.com/jcredland/jpm/pull/1/files#r47277812>.
>
> — Reply to this email directly or view it on GitHub[3].


>



Links:

  1. https://github.com/jcredland/jpm/pull/1#discussion_r47280364
  2. https://github.com/jcredland/jpm/pull/1#discussion_r47277812
  3. https://github.com/jcredland/jpm/pull/1/files#r47280364